crates/oci/src/client.rs

```diff
@@ -84,6 +84,17 @@ pub struct ClientOpts {
     pub content_ref_inline_max_size: usize,
 }
 
+/// Controls whether predefined annotations are generated when pushing an application.
+/// If an explicit annotation has the same name as a predefined one, the explicit
+/// one takes precedence.
+#[derive(Debug, PartialEq)]
+pub enum PredefinedAnnotations {
```

Comment:
I wonder if adding this, or something similar, into ClientOpts might be an option?  Then we'd be able to access it from the client when we need it and wouldn't have to pass it through functions as an additional arg.

Reply:
Folding into ClientOpts definitely feels worth investigation.  That would probably allow us to get rid of the enum and use a boolean into the bargain.  I'll take a look.

Reply:
@vdice I had an explore of the ClientOpts approach.  It ended up feeling a bit awkward that the explicit annotations were set up in one place and the inferred ones in another.  So I am going to back that out.  Appreciate the idea though!